packages/rocketchat-api/server/v1/users.js

```diff
@@ -402,7 +402,7 @@ API.v1.addRoute('users.createToken', { authRequired: true }, {
 API.v1.addRoute('users.getPreferences', { authRequired: true }, {
 	get() {
 		const user = Users.findOneById(this.userId);
-		if (user.settings) {
+		if (user.settings && !_.isEmpty(user.settings)) {
 			const { preferences } = user.settings;
```

Comment:
Your solution doesn't solve the problem because if the `user.settings` is not empty, containing any other setting, but doesn't contains the `preferences` property, it will still fail with the same error of trying to access `language` of undefined.

```suggestion
			const { preferences = {} } = user.settings;
```